frontend/src/employee-frontend/components/absences/AbsenceModal.tsx

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React from 'react'
+import { AbsenceType } from 'lib-common/generated/enums'
+import {
+  AbsenceTypes,
+  CareTypeCategories,
+  CareTypeCategory
+} from '../../types/absence'
+import { useTranslation } from '../../state/i18n'
+import FormModal from 'lib-components/molecules/modals/FormModal'
+import { FixedSpaceColumn } from 'lib-components/layout/flex-helpers'
+import { Label, LabelText } from '../common/styled/common'
+import Radio from 'lib-components/atoms/form/Radio'
+import Checkbox from 'lib-components/atoms/form/Checkbox'
+
+export default React.memo(function AbsenceModal({
+  show,
+  onSave,
+  saveDisabled,
+  onCancel,
+  selectedAbsenceType,
+  setSelectedAbsenceType,
+  selectedCareTypeCategories,
+  updateCareTypes
+}: {
+  show: boolean
+  onSave: () => void
+  saveDisabled: boolean
+  onCancel: () => void
+  selectedAbsenceType: AbsenceType | null
+  setSelectedAbsenceType: (value: AbsenceType | null) => void
+  selectedCareTypeCategories: CareTypeCategory[]
+  updateCareTypes: (value: CareTypeCategory) => void
+}) {
+  const { i18n } = useTranslation()
+  return show ? (
```

Comment:
In react-world, I prefer short circuit evaluation `show &&` instead of ternaries.

Reply:
Tbh I would leave the `show` out of props and just short circuit eval in the parent container `showModal && <AbsenceModal ..`

Reply:
This works in the parent but not here, because a component cannot return `false`.

Reply:
Right, you'd have to wrap it in a fragment. Anyway, I think it's unidiomatic to pass `show` to the modal.

Reply:
You're right, fixed!